src/fft_mkl.f90

```diff
@@ -422,73 +500,70 @@ subroutine fft_3d_c2c(in, out, isign)
          !       else if (isign==DECOMP_2D_FFT_BACKWARD) then
          !          status = DftiComputeBackward(c2c_z, in(:,1,1), wk1(:,1,1))
          !       end if
-#ifdef OVERWRITE
-         status = wrapper_c2c_inplace(c2c_z, in, isign)
-#else
-         call alloc_z(wk1, ph)
-         status = wrapper_c2c(c2c_z, in, wk1, isign)
-#endif
+         if (inplace) then
+            status = wrapper_c2c_inplace(c2c_z, in, isign)
+         else
+            call alloc_z(wk1, ph)
+            status = wrapper_c2c(c2c_z, in, wk1, isign)
+         end if
          if (status /= 0) call decomp_2d_abort(__FILE__, __LINE__, status, "wrapper_c2c")
 
          ! ===== Swap Z --> Y =====
-#ifdef OVERWRITE
-         call transpose_z_to_y(in, wk2_c2c, ph)
-#else
-         call transpose_z_to_y(wk1, wk2_c2c, ph)
-#endif
+         if (inplace) then
+            call transpose_z_to_y(in, wk2_c2c, ph)
+         else
+            call transpose_z_to_y(wk1, wk2_c2c, ph)
+         end if
 
          ! ===== 1D FFTs in Y =====
-#ifdef OVERWRITE
-         do k = 1, ph%xsz(3)
-            status = wrapper_c2c_inplace(c2c_y, wk2_c2c(1, 1, k), isign)
-            if (status /= 0) call decomp_2d_abort(__FILE__, __LINE__, status, "wrapper_c2c")
-         end do
-#else
-         call alloc_y(wk2b, ph)
-         do k = 1, ph%xsz(3) ! one Z-plane at a time
-            !          if (isign==DECOMP_2D_FFT_FORWARD) then
-            !             status = DftiComputeForward(c2c_y, wk2(:,1,k), wk2b(:,1,k))
-            !          else if (isign==DECOMP_2D_FFT_BACKWARD) then
-            !             status = DftiComputeBackward(c2c_y, wk2(:,1,k), wk2b(:,1,k))
-            !          end if
-            status = wrapper_c2c(c2c_y, wk2_c2c(1, 1, k), wk2b(1, 1, k), isign)
-            if (status /= 0) call decomp_2d_abort(__FILE__, __LINE__, status, "wrapper_c2c")
-         end do
-#endif
+         if (inplace) then
+            do k = 1, ph%xsz(3)
+               status = wrapper_c2c_inplace(c2c_y, wk2_c2c(:, :, k), isign)
```

Comment:
Are these changes correcting errors? The `wrapper_c2c_inplace` doesn't seem to have changed?

Reply:
I think this modification comes from the removal of the `#ifdef OVERWRITE` preprocessor.